src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -2140,6 +2140,7 @@ static void updateAttrInNotifyCer
           caP->previousValue = new ContextAttribute();
         }
 
+        caP->previousValue->type        = caP->type;
         caP->previousValue->valueType   = caP->valueType;
         caP->previousValue->stringValue = caP->stringValue;
         caP->previousValue->boolValue   = caP->boolValue;
```

Comment:
Just above `if (targetAttr->type != "")` in L2175 include:

- An empty line to separate from the above block
- This comment

> /* Set attribute type (except if new value is "", which means that the type is not going to change) */